### PR TITLE
Processor doc: add description for queued processor

### PR DESCRIPTION
**Pull request by huikang**
Signed-off-by: Hui Kang <kangh@us.ibm.com>

### CODE REVIEW

File: processor/README.md [hunk 1]
```diff
@@ -194,7 +194,28 @@ examples on using the processor.
 <FILL ME IN - I'M LONELY!>
 
 ## <a name="queued"></a>Queued Processor
-<FILL ME IN - I'M LONELY!>
+The queued processor uses a ring buffer queue to relay trace data from the
+receiver or previous processor to the next span processor. On
+receiving trace data, the queued processor produces items to the queue. At the
+same time, the queued processor has one or more workers which consume the trace
+data in the queue by sending them to the next span processor.
```
Comment (rghetia) [inline processor/README.md]: maybe add some description about retry logic?

Comment (huikang) [reply processor/README.md]: Updated. Thanks for your feedback.

File: processor/README.md [hunk 2]
```diff
@@ -194,7 +194,30 @@ examples on using the processor.
 <FILL ME IN - I'M LONELY!>
 
 ## <a name="queued"></a>Queued Processor
-<FILL ME IN - I'M LONELY!>
+The queued processor uses a ring buffer queue to relay trace data from the
+receiver or previous processor to the next span processor. On
```
Comment (pjanotti) [inline processor/README.md]:
```suggestion
receiver or previous processor to the next processor. On
```

File: processor/README.md [hunk 3]
```diff
@@ -194,7 +194,30 @@ examples on using the processor.
 <FILL ME IN - I'M LONELY!>
 
 ## <a name="queued"></a>Queued Processor
```
Comment (pjanotti) [inline processor/README.md]: Not from your change, but a suggestion here:
```suggestion
## <a name="queued"></a>Queued Retry Processor
```

File: processor/README.md [hunk 4]
```diff
@@ -194,7 +194,30 @@ examples on using the processor.
...
-<FILL ME IN - I'M LONELY!>
+The queued processor uses a ring buffer queue to relay trace data from the
```
Comment (pjanotti) [inline processor/README.md]:
```suggestion
The queued retry processor uses a bounded queue to relay trace data from the
```

File: processor/README.md [hunk 5]
```diff
@@ -194,7 +194,30 @@ examples on using the processor.
...
+receiver or previous processor to the next span processor. On
+receiving trace data, the queued processor produces items to the queue. At the
```
Comment (pjanotti) [inline processor/README.md]:
```suggestion
Received trace data is enqueued immediately If the queue is not full . At the
```

File: processor/README.md [hunk 6]
```diff
@@ -194,7 +194,30 @@ examples on using the processor.
...
+same time, the queued processor has one or more workers which consume the trace
```
Comment (pjanotti) [inline processor/README.md]:
```suggestion
same time, the processor has one or more workers which consume the trace
```

File: processor/README.md [hunk 7]
```diff
@@ -194,7 +194,30 @@ examples on using the processor.
...
+data in the queue by sending them to the next span processor. On failure of
```
Comment (pjanotti) [inline processor/README.md]:
```suggestion
data in the queue by sending them to the next processor. On failure of
```

**Comment by huikang** [comment]
@pjanotti , updated. Thanks for the suggestions.

File: processor/README.md [hunk 8]
```diff
@@ -193,8 +193,31 @@ examples on using the processor.
...
 
-## <a name="queued"></a>Queued Processor
-<FILL ME IN - I'M LONELY!>
+## <a name="queued"></a>Queued Retry Processor
+The queued retry processor uses a bounded queue to relay trace data from the
+receiver or previous processor to the next processor.
+Received trace data is enqueued immediately if the queue is not full . At the
+same time, the processor has one or more workers which consume the trace
+data in the queue by sending them to the next processor. On failure of
+consuming the trace data, the processor retries after some backoff delay
```
Comment (tigrannajaryan) [inline processor/README.md]: Unclear what "consuming" refers to. Maybe say instead: "If relaying the trace data to the next processor or exporter in the pipeline fails ..."

Comment (huikang) [reply processor/README.md]: Updated. Thanks.

**Pull request merged**